model/artifact/db.go

```diff
@@ -115,37 +116,36 @@ func (e Entry) Upsert() error {
 }
 
 func (e Entry) Update() error {
-	err := db.Update(
-		Collection,
-		bson.M{
-			TaskIdKey:    e.TaskId,
-			TaskNameKey:  e.TaskDisplayName,
-			BuildIdKey:   e.BuildId,
-			ExecutionKey: e.Execution,
-		},
-		bson.M{
-			"$set": bson.M{
-				FilesKey: e.Files,
-			},
-			"$setOnInsert": bson.M{
-				ExecutionKey: e.Execution,
-			},
-		},
-	)
-	if err != nil && e.Execution == 0 {
+	var err error
+	if e.Execution == 0 {
 		err = db.Update(
 			Collection,
 			bson.M{
 				TaskIdKey:   e.TaskId,
 				TaskNameKey: e.TaskDisplayName,
 				BuildIdKey:  e.BuildId,
+				"$or": []bson.M{
```

Comment:
The only difference between these two queries _is_ the update. We can set this bson.M document at the top like 
```
update := bson.M{
	TaskIdKey:   e.TaskId,
	TaskNameKey: e.TaskDisplayName,
	BuildIdKey:  e.BuildId
}
```
and then say 
```
if e.Execution == 0 {
    update["$or"] = []bson.M{ <execution stuff > }
}
```
Then we can just call db.Update(Collection, update, etc) outside of the if statement. This would prevent the other code duplication.

Reply:
wow youre so smart! done